### PR TITLE
recast / detour upgrade

### PR DESCRIPTION
**Pull request by WyrdanGames**
Upgrade to recast / detour version from 12/2022 with additional navigation tests (should work with old version also)

There have been some bugfixes and updates in the last few years

### CODE REVIEW

File: Source/ThirdParty/Recast/Source/RecastMeshDetail.cpp [hunk 1]
```diff
@@ -653,8 +653,8 @@ static bool buildPolyDetail(rcContext* ctx, const float* in, const int nin,
 	for (int i = 0; i < nin; ++i)
 		rcVcopy(&verts[i*3], &in[i*3]);
 	
-	edges.resize(0);
-	tris.resize(0);
+	edges.clear();
+	tris.clear();
```
Comment (rokups) [inline Source/ThirdParty/Recast/Source/RecastMeshDetail.cpp:657]: I wonder why they did this all over the place. `clear()` frees backing storage so memory reallocation will happen, while `resize(0)` keeps backing storage which is reused when new stuff is added to the container. I somewhat doubt there were memory usage issues...

Comment (eugeneko) [reply Source/ThirdParty/Recast/Source/RecastMeshDetail.cpp:657]: > clear() frees backing storage so memory reallocation will happen

Does it?

**Comment by eugeneko** [comment]
@rokups we probably want to fork Detour/Recast as well because we patch it too
I think we can do that afterwards.

@WyrdanGames can you please link the commit of Detour/Recast you used?

**Comment by WyrdanGames** [comment]
https://github.com/recastnavigation/recastnavigation/tree/d0b2ed8ff5d371f375cf49dad7ca96e8b22142d0

File: Source/ThirdParty/Recast/Source/RecastMeshDetail.cpp (hunk 1 above)
Comment (eugeneko) [reply Source/ThirdParty/Recast/Source/RecastMeshDetail.cpp:657]: How it's all related to storage?

Comment (rokups) [reply Source/ThirdParty/Recast/Source/RecastMeshDetail.cpp:657]: Eh... I mistakenly assumed this is backed by `std::vector<>`. Their custom containers do not free underlying memory in both cases. That explains the change

**Comment by eugeneko** [comment]
@WyrdanGames FYI: if you ever consider updating 3rdparties again -- check if there's repo in rbfx org and ping us if there's none.
For patched 3rdparties we keep forks, it makes updating easier. You just rebase "rbfx" branch and fix conflicts if any...

**Review by eugeneko — APPROVED** [review]

**Pull request merged**